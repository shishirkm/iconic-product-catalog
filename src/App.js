import React, { useEffect } from 'react';
import Header from './components/Header';
import { useDispatch } from 'react-redux';
import { fetchProducts } from './reducers/productListReducer';
import ProductList from './components/ProductList';
import ProductFilter from './components/ProductFilter';
import SC from './AppStyles';
import ProductSearch from './components/ProductSearch';
import ProductSort from './components/ProductSort';

const App = () => {

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchProducts);
  }, []);
  
  return (
    <SC.Wrapper className="App">
      <Header />
      <SC.Row>
        <SC.Filter>
          <ProductFilter />
        </SC.Filter>
        <SC.ProductList>
          <SC.SortSearchWrapper>
            <ProductSearch />
            <ProductSort />
          </SC.SortSearchWrapper>
          <ProductList />
        </SC.ProductList>
      </SC.Row>
    </SC.Wrapper>
  );
}

export default App;
