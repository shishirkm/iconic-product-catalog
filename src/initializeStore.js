import thunk from "redux-thunk";

const { combineReducers, createStore, applyMiddleware } = require("redux");
const { productListReducer } = require("./reducers/productListReducer");

const reducer = combineReducers({
  ProductListReducer: productListReducer,
});

const initializeStore = () => {
    const middlewares = [thunk];
if (process.env.NODE_ENV === "development") {
  const { createLogger } = require("redux-logger");
  middlewares.push(
    createLogger({ collapsed: true, duration: true, diff: true })
  );
}
const store = createStore(reducer, applyMiddleware(...middlewares));
return store;
}

export default initializeStore;
