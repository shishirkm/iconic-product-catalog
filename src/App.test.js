import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";
import { Provider } from "react-redux";
import App from "./App";
import initializeStore from "./initializeStore";

it("renders without crashing", () => {
  const mockStore = initializeStore();

  const wrapper = mount(
    <Provider store={mockStore}>
      <App />
    </Provider>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
