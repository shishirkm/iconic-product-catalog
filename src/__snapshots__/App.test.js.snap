// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App>
    <styled.div
      className="App"
    >
      <div
        className="sc-iqHYGH bBtsme App"
      >
        <Header>
          <styled.div>
            <div
              className="sc-bdfBwQ enELhY"
            >
              <div
                className="header-text"
              >
                THE ICONIC
              </div>
            </div>
          </styled.div>
        </Header>
        <styled.div>
          <div
            className="sc-crrsfI fBeTKC"
          >
            <styled.div>
              <div
                className="sc-dQppl jVREIy"
              >
                <ProductFilter>
                  <styled.div>
                    <div
                      className="sc-pFZIQ"
                    >
                      <styled.div
                        className="label"
                      >
                        <div
                          className="sc-jrAGrp iPZhPA label"
                        >
                          Category
                        </div>
                      </styled.div>
                    </div>
                  </styled.div>
                </ProductFilter>
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-bqyKva kYYQjz"
              >
                <styled.div>
                  <div
                    className="sc-kstrdz UYgsY"
                  >
                    <ProductSearch>
                      <styled.div>
                        <div
                          className="sc-hBEYos gngunG"
                        >
                          <input
                            onChange={[Function]}
                            placeholder="Search by name"
                            type="text"
                          />
                        </div>
                      </styled.div>
                    </ProductSearch>
                    <ProductSort>
                      <styled.div>
                        <div
                          className="sc-fodVxV iPgqII"
                        >
                          <select
                            id="sort-products"
                            name="sort"
                            onChange={[Function]}
                          >
                            <option
                              key="popularity"
                              value="popularity"
                            >
                              Popularity
                            </option>
                            <option
                              key="priceDesc"
                              value="priceDesc"
                            >
                              Price High to Low
                            </option>
                            <option
                              key="priceAsc"
                              value="priceAsc"
                            >
                              Price Low to High
                            </option>
                            <option
                              key="brandAsc"
                              value="brandAsc"
                            >
                              Brand A to Z
                            </option>
                            <option
                              key="brandDesc"
                              value="brandDesc"
                            >
                              Brand Z to A
                            </option>
                          </select>
                        </div>
                      </styled.div>
                    </ProductSort>
                  </div>
                </styled.div>
                <ProductList>
                  <styled.div>
                    <div
                      className="sc-iBPRYJ knmzvq"
                    >
                      <styled.div>
                        <div
                          className="sc-fubCfw dMXtRu"
                        />
                      </styled.div>
                    </div>
                  </styled.div>
                </ProductList>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </App>
</Provider>
`;
